google-oauth-client/src/main/java/com/google/api/client/auth/openidconnect/IdTokenVerifier.java

```diff
@@ -280,5 +402,163 @@ public Builder setAcceptableTimeSkewSeconds(long acceptableTimeSkewSeconds) {
       this.acceptableTimeSkewSeconds = acceptableTimeSkewSeconds;
       return this;
     }
+
+    /** Returns an instance of the {@link Environment} */
+    public final Environment getEnvironment() {
+      return environment;
+    }
+
+    /** Sets the environment. Used mostly for testing */
+    public Builder setEnvironment(Environment environment) {
+      this.environment = environment;
+      return this;
+    }
+
+    /**
+     * Set the HttpTransportFactory used for requesting public keys from the certificate URL. Used
+     * mostly for testing.
+     *
+     * @param httpTransportFactory the HttpTransportFactory used to build certificate URL requests
+     * @return the builder
+     */
+    public Builder setHttpTransportFactory(HttpTransportFactory httpTransportFactory) {
```

Comment:
> Used mostly for testing.

Would you check whether this needs to be public?

Reply:
We don't appear to have a default http transport, so the user needs to set one. We could potentially set a default (using java.net transport). This is also important for picking which http transport library to use (and configuring proxies and such)

Reply:
Good catch. If I understand correctly, without a default, the existing code all fails with NPE. That's not good. I want this change to work for most (expected) use cases without any code modification.

Reply:
I will set to nettransport by default. And I will add extended error message in case cert failed to fetch pointing to transport and a variable to disable

Reply:
Done, except current interface does not let us hav exception message, so it goes to docs and logger